components/observations/uploader/ObservationsUploader.ts

```diff
@@ -172,6 +340,22 @@ export class ObservationUploader {
     this.checkInitialized();
     this.subscribers = this.subscribers.filter(subscriber => subscriber !== callback);
   }
+
+  getState(): UploaderState {
+    // The `map` call is because tsc needs a little help to grok the type of the filter expression
```

Comment:
I've always just used normal `for` loops since the type narrowing works for those (and there's no performance penalty AFAICT)